### PR TITLE
CSA - 96: Add RoadmapPage

### PR DESCRIPTION
**Pull request by VovaLyapota**

### CODE REVIEW

File: src/components/ui/Container/Container.module.css [hunk 1]
```diff
@@ -5,14 +5,22 @@
...
     padding: 0 32px;
   }
 }
+
 @media screen and (min-width: 1440px) {
   .wrapper {
+    margin-top: 40px;
+    margin: 0 auto;
+    width: 100%;
     max-width: 1440px;
```
Comment (Tanyapavliuk) [inline src/components/ui/Container/Container.module.css:19]: No need to duplicate properties with the same values because they are inherited

File: src/components/Roadmap/Roadmap/Roadmap.module.css [hunk 2]
```diff
@@ -0,0 +1,107 @@
+.wrapper {
+  padding-top: 40px;
+  padding-bottom: 40px;
+}
+
+.title {
+  position: relative;
+
+  margin: 0 auto 24px;
```
Comment (Tanyapavliuk) [inline src/components/Roadmap/Roadmap/Roadmap.module.css]: Why did you add a space?

File: src/components/GuideList/GuideList.module.css [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  padding-bottom: 32px;
+}
+
+.container {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+}
+
+.title {
```
Comment (tymnik) [inline src/components/GuideList/GuideList.module.css:13]: We have components PageHeading and PageText for h2 and p with most styles

File: src/components/Layout/Layout.tsx [hunk 4]
```diff
@@ -15,7 +14,6 @@ import styles from './LayoutFallback.module.css';
 const Layout = () => {
   const location = useLocation();
   const dispatch = useAppDispatch();
-  const navigate = useNavigate();
   const { id, refresh } = useAppSelector(selectToken);
   const isLoggedIn = useAppSelector(selectIsLoggedIn);
   const pathsWithoutFooter = ['/register', '/login'];
```
Comment (tymnik) [inline src/components/Layout/Layout.tsx:19]: can we use here our constants from routes.ts?

File: src/components/GuideList/GuideList.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+}[];
+
+interface GuideListProps {
+  title: string;
+  array: DataArray;
+}
+const itemStyles = [styles.first, styles.second, styles.third];
+
+const GuideList = ({ title, array }: GuideListProps) => {
+  return (
```
Comment (StanislavKharkiv) [inline src/components/GuideList/GuideList.tsx]: I look at this props and don't understand what is `array`. It's can be anything. Can we change name? probably `data` or something else ?

File: src/components/RoadmapPageContent/Roadmap/Roadmap.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+const Roadmap = ({ title, roadmapSteps }: RoadmapProps) => {
+  const [openedDropdownId, setOpenedDropdownId] = useState<number | null>(null);
+  const [modalContent, setModalContent] = useState<Vertex | null>(null);
+  const { mainStep, subSteps } = splitVertices(roadmapSteps);
+
+  const handleModalVisibilityContent = (step: Vertex) => () =>
+    setModalContent((p) => {
+      return p === null ? step : null;
+    });
+
```
Comment (StanislavKharkiv) [inline src/components/RoadmapPageContent/Roadmap/Roadmap.tsx]: I think we more clear make 2 sort functions () =>  setModalContent and clear content

File: src/components/RoadmapPageContent/RoadmapProgress/RoadmapProgress.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+export const RoadmapProgress = () => {
+  const isLoggedIn = useAppSelector(selectIsLoggedIn);
+
+  return (
+    <div className={styles.container}>
+      <h3 className={styles.title}>Your progress:</h3>
+      <div className={styles.defaultBar} aria-hidden="true" />
+      {isLoggedIn ? null : (
+        <p className={styles.text}>
```
Comment (StanislavKharkiv) [inline src/components/RoadmapPageContent/RoadmapProgress/RoadmapProgress.tsx]: Probably we can just write `!isLoggedIn && (...)`

File: src/components/modals/RoadmapStepModal/RoadmapStepModal.tsx [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+      <p className={styles.modalTitle}>{step.info_card.title}</p>
+      <p className={styles.description}>
+        {step.info_card.description}
+        {step.info_card.links.length > 0 ? (
+          <>
+            <br />
+            Visit the following resources to learn more:
+          </>
+        ) : null}
+      </p>
```
Comment (StanislavKharkiv) [inline src/components/modals/RoadmapStepModal/RoadmapStepModal.tsx]: the same, just use `&&` operator instead `?`

**Review by StanislavKharkiv — APPROVED** [review]
